### ISSUE TITLE
Time Scaled training does not respect the "Frame to Skip" and/or "Wait Time" parameters

### ISSUE THREAD

**Issue by gururise** [Unity-Technologies/ml-agents#403]
tldr: The simulation behaves differently when Time Scale is set to 1 vs 100.

I have an environment that requires the use of the **Frame to Skip** and **Wait Time** in the Academy in order to generate movement at intervals rather than every frame.  I'm currently using Frame To Skip (50) and Wait Time (0); however I see the same problem using Frame to Skip (3) and Wait Time (0.25).  I am using these parameters to prevent my agent (which utilizes physics and rigid bodies) from moving its motors every time step, rather, when working properly, the agent should act (ie. AgentStep() ) once every 50 frames.

This configuration runs and trains fine if the Training Configuration->Time Scale is set to 1; however, setting it to 100 (to accelerate training) makes the physics engine behave weird and the results are no longer consistent (or the same) in comparison to training at Time Scale of 1.

By setting the time scale to 100, the game engine seems to ignore Frame to Skip and/or the Wait Time setting and cause the Agent to act at every timestep during training.  This causes physics instability in my environment.

**Comment by vincentpierre** [comment]
The way ML-Agents works is as follows : `WaitTime` only works at inference. Frame skip is the number of steps between each decision (call to the brain) but the action happens at every step (AgentStep).
In master there is no way to act at specific steps as of version ML-Agents Beta 0.2.1d. There is a new feature called On Demand Decision in 0.3 which allows you to fully control when the agent decides and act. If you are interested, check out the branch `development-0.3`, the documentation is not quite there yet but you should be able to call `RequestDecision` and `RequestAction` on your agent on that branch.

**Comment by mmattar** [comment]
Hi @gururise - closing this out due to inactivity, but please reopen or file a new issue if you need any additional support.
